plugins/libplugin.c

```diff
@@ -480,12 +497,49 @@ static struct command_result *handle_init(struct command *init_cmd,
 					  ".allow-deprecated-apis"),
 				"true");
 
+	opttok = json_get_member(buf, params, "options");
+	json_for_each_obj(i, t, opttok) {
+		char *opt = json_strdup(NULL, buf, t);
+		for (size_t i = 0; i < tal_count(opts); i++) {
+			char *problem;
+			if (!streq(opts[i].name, opt))
+				continue;
+			problem = opts[i].handle(json_strdup(opt, buf, t+1),
+						 opts[i].arg);
+			if (problem)
+				plugin_err("option '%s': %s",
+					   opts[i].name, problem);
+			break;
+		}
+		tal_free(opt);
+	}
+
 	if (init)
 		init(init_cmd->rpc);
 
 	return command_done_ok(init_cmd, "");
 }
 
+char *u64_option(const char *arg, u64 *i)
```

Comment:
Almost same code (with same "This is how the manpage says to do it.  Yech.") exists in `lightningd/options.c` under function `opt_set_s32`, perhaps unifying this code in some `common/` library is better?

Reply:
I should probably move the u32/u64 etc options into ccan, since it has all the int/uint/long/ulong values already.